Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynRuleSetGeneratorTests.cs

```diff
@@ -225,51 +194,73 @@ public void RoslynRuleSet_Common_Parameters()
-        {
-            { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
-        }));
-        var rules = new[]
-        {
-            new SonarRule("csharpsquid", "active1", true),
-        };
-
-        new Action(() => generator.Generate("cs", rules)).Should()
+        var context = new RuleSetTestContext(
+            [new("csharpsquid", "active1", true)],
+            new Dictionary<string, string>
+            {
+                { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
+            });
+
+        new Action(() => context.GenerateRuleSet()).Should()
             .ThrowExactly<AnalysisException>().WithMessage("Property does not exist: sonaranalyzer-cs.analyzerId. This property should be set by the plugin in SonarQube.");
     }
 
     [TestMethod]
     public void RoslynRuleSet_RuleNamespace_Proprety_Missing()
     {
-        var generator = new RoslynRuleSetGenerator(new ListPropertiesProvider(new Dictionary<string, string>
-        {
-            { "sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp" },
-        }));
-
-        var rules = new[]
-        {
-            new SonarRule("csharpsquid", "active1", true),
-        };
-
-        new Action(() => generator.Generate("cs", rules)).Should()
+        var context = new RuleSetTestContext(
+            [new("csharpsquid", "active1", true)],
+            new Dictionary<string, string>
+            {
+                { "sonaranalyzer-cs.analyzerId", "SonarAnalyzer.CSharp" },
+            });
+
+        new Action(() => context.GenerateRuleSet()).Should()
             .ThrowExactly<AnalysisException>().WithMessage("Property does not exist: sonaranalyzer-cs.ruleNamespace. This property should be set by the plugin in SonarQube.");
     }
 
     [TestMethod]
     public void RoslynRuleSet_PropertyName_IsCaseSensitive()
     {
-        var generator = new RoslynRuleSetGenerator(new ListPropertiesProvider(new Dictionary<string, string>
-        {
+        var context = new RuleSetTestContext(
+            [new("csharpsquid", "active1", true)],
+            new Dictionary<string, string>
+            {
             { "sonaranalyzer-cs.ANALYZERId", "SonarAnalyzer.CSharp" },
             { "sonaranalyzer-cs.ruleNamespace", "SonarAnalyzer.CSharp" },
-        }));
-
-        var rules = new[]
-        {
-            new SonarRule("csharpsquid", "active1", true),
-        };
+            });
 
-        new Action(() => generator.Generate("cs", rules)).Should()
+        new Action(() => context.GenerateRuleSet()).Should()
             .ThrowExactly<AnalysisException>().WithMessage("Property does not exist: sonaranalyzer-cs.analyzerId. This property should be set by the plugin in SonarQube.");
     }
+
+    private class RuleSetTestContext
+    {
+        private RoslynRuleSetGenerator RuleSetGenerator { get; }
+        private List<SonarRule> Rules { get; }
+        private string Language { get; }
```

Comment:
Private property doesn't make sense
```suggestion
        private readonly RoslynRuleSetGenerator ruleSetGenerator;
        private readonly List<SonarRule> rules;
        private readonly string language;
```